pkg/handler/handler.go

```diff
@@ -230,7 +238,12 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 					}
 
 					// serve the first matching media type in the sorted clause list
-					data, etag, lastModified := accepts.GetDataAndETag()
+					data, etag, lastModified, err := accepts.GetDataAndETag()
+					if err != nil {
+						klog.Errorf("Error in OpenAPI handler: %s", err)
+						w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Instead of returning a 500, the old behavior was to return the previous spec when the server fails to serialize a new spec. I'm not sure if it ever occurs in reality, but is there a reason to change the behavior?

Reply:
@DangerOnTheRanger We can probably initialize the cache with the previous value, so that if it fails, we continue to have the value we had before, what do you think?